soccer/motion/MotionControl.cpp

```diff
@@ -178,45 +175,40 @@ void MotionControl::run() {
 
 
 		//	evaluate path - where should we be right now?
-		Point targetPos;
-		bool pathValidNow = _robot->path()->evaluate(
-			timeIntoPath,
-			targetPos,
-			targetVel);
+		bool pathValidNow = _robot->path()->evaluate(timeIntoPath, target);
 		if (!pathValidNow) {
-			targetVel.x = 0;
-			targetVel.y = 0;
+			target.vel = Geometry2d::Point();
 		}
 		//	tracking error
-		Point posError = targetPos - _robot->pos;
+		Point posError = target.pos - _robot->pos;
 
 		//	acceleration factor
-		Point nextTargetVel, _;
-		_robot->path()->evaluate(timeIntoPath + 1.0/60.0, _, nextTargetVel);
-		Point acceleration = (nextTargetVel - targetVel) / 60.0f;
+		Planning::MotionInstant nextTarget;
+		_robot->path()->evaluate(timeIntoPath + 1.0 / 60.0, nextTarget);
+		Point acceleration = (nextTarget.vel - target.vel) / 60.0f;
```

Comment:
Yep, you're right.  Not sure why I thought it was rotated wrong before, it looks just fine :)

I do agree it needs some work, but I think the accel stuff here is fairly sensible.  It's checking how much acceleration is being specified by the path and giving the robot velocity command a boost to compensate.  Otherwise, the acceleration wouldn't be handled at all until error builds up and gets multiplied by the P term.  Not saying this is an optimal way to solve the problem, but it works for now until we overhaul everything.